.github/scripts/cmd/validator/validator.go

```diff
@@ -0,0 +1,196 @@
+package validator
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/coinhall/yacar/internal/enums"
+	"github.com/coinhall/yacarsdk/v2"
+)
+
+func Start(filePaths []string) {
+	validateYacarJSONs(filePaths)
+	log.Println("Validated JSONs successfully...")
+}
+
+func validateYacarJSONs(filePaths []string) {
+	var wg sync.WaitGroup
+	for _, filePath := range filePaths {
+		wg.Add(1)
+		go func(filePath string) {
+			defer wg.Done()
+
+			file, err := os.Open(filePath)
+			if err != nil {
+				log.Fatalf("error while opening file: %s", err)
+			}
+			defer file.Close()
+
+			if err := validateYacarJSON(filePath, file); err != nil {
+				log.Fatalf("%s\npath: %s", err, filePath)
+			}
+
+		}(filePath)
+	}
+	wg.Wait()
+}
+
+func validateYacarJSON(filePath string, file *os.File) error {
```

Comment:
Arguments feel convoluted as `file` can be derived from `filePath`, which raises the question - why is the file opened if the file path is "unknown". Perhaps a better flow would be to open only if the file path is recognized.